Public/Src/Cache/DistributedCache.Host/Service/Internal/DualServerClientContentStore.cs

```diff
@@ -0,0 +1,172 @@
+
+namespace BuildXL.Cache.Host.Service.Internal
+{
+    /// <summary>
+    /// Content store containing both a distributed <see cref="LocalContentStore"/> and <see cref="ServiceClientContentStore"/>.
+    /// </summary>
+    public class DualServerClientContentStore : IContentStore
+    {
+        private LocalContentServer _localContentServer;
+        private ServiceClientContentStore _serviceClientContentStore;
+
+        /// <inheritdoc />
+        public bool StartupCompleted => (_localContentServer?.StartupCompleted ?? false) && (_serviceClientContentStore?.StartupCompleted ?? false);
+
+        /// <inheritdoc />
+        public bool StartupStarted => (_localContentServer?.StartupStarted ?? false) && (_serviceClientContentStore?.StartupStarted ?? false);
+
+        /// <inheritdoc />
+        public bool ShutdownCompleted => (_localContentServer?.ShutdownCompleted ?? false) && (_serviceClientContentStore?.ShutdownCompleted ?? false);
+
+        /// <inheritdoc />
+        public bool ShutdownStarted => (_localContentServer?.ShutdownStarted ?? false) && (_serviceClientContentStore?.ShutdownStarted ?? false);
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="fileSystem">File system object used by the client.</param>
+        /// <param name="logger">Logger.</param>
+        /// <param name="copier">File copier used by the content server.</param>
+        /// <param name="pathTransformer">File path transformer used by the content server.</param>
+        /// <param name="maxQuotaMB">Maximum size (in MB) of the cache.</param>
+        /// <param name="cacheName">Name of the cache.</param>
+        /// <param name="grpcPort">GRPC port which the server listens to and the client transmits to.</param>
+        /// <param name="stampId">Redis stamp identifier.</param>
+        /// <param name="ringId">Redis ring identifier.</param>
+        /// <param name="scenarioName">Unique name for the CAS. Allows multiple CAS services to coexist.</param>
+        /// <param name="dataRootPath">Path to the root of CAS content.</param>
+        public DualServerClientContentStore(
+            IAbsFileSystem fileSystem,
+            ILogger logger,
+            IAbsolutePathFileCopier copier,
+            IAbsolutePathTransformer pathTransformer,
+            int maxQuotaMB,
+            string cacheName,
+            int grpcPort,
+            string stampId,
+            string ringId,
+            string scenarioName,
+            AbsolutePath dataRootPath)
+        {
+            Contract.Assert(grpcPort >= 0);
+            Contract.Assert(maxQuotaMB >= 0);
+            Contract.Assert(!string.IsNullOrWhiteSpace(cacheName));
+            Contract.Assert(!string.IsNullOrWhiteSpace(ringId));
+            Contract.Assert(!string.IsNullOrWhiteSpace(stampId));
+
+            var redisConnectionString = Environment.GetEnvironmentVariable(EnvironmentConnectionStringProvider.RedisConnectionStringEnvironmentVariable);

```

Comment:
As I suggested already, I think it make sense to create a configuration type and put this value directly there instead of reading environment variables in the constructor.